netlify/edge-functions/subdomain-redirect.ts

```diff
@@ -63,9 +63,8 @@ async function redirectNetlifySubdomains(
   console.log(`URL is on allow list: ${isOnAllowList}`);
 
   if (subdomain && !redirected && !isOnAllowList) {
-    const redirectTargetUrl = new URL(
-      `https://${subdomain}.netlify.thenational.academy/`,
-    ).href;
+    const redirectTargetUrl = new URL(request.url);
+    redirectTargetUrl.host = `${subdomain}.netlify.thenational.academy`;
```

Comment:
Ok! This was the issue all along.

Netlify expect you to either let them manage your DNS or to [delegate a sub-domain for them to manage for deploy previews](https://docs.netlify.com/domains-https/custom-domains/automatic-deploy-subdomains/#:~:text=setup%20steps.-,Want%20to%20use%20a%20subdomain%20without%20bringing%20the%20apex%20domain%20to%20Netlify%20DNS%3F,-You%20can%20bring).

☝🏻 we're not doing that, instead we've rolled our own solution in Cloudflare of wildcard DNS and the subdomain-redirect edge function (above).

This means that internally Netlify is unaware of the `netlify.thenational.academy` sub-domain on deploy previews. Instead it treats all requests as being made to `netlify.app` — which is why the redirects inside the Clerk middleware were always made through `netlify.app`. This redirect would trigger the above subdomain-redirect edge function. 

This function however was severely limited! It would always redirect to the root of the app, discarding both the pathname and search params.

My change above alters the behaviour of the subdomain redirect function (which is only applied to deploy previews). It now only rewrite the host portion of the URL,  preserving the pathname and search params.

This means that deploy previews go through an additional redirect, but eventually settle on the proper domain and everything gets where it needs to go.

Ideally we'd do the proper Netlify way to stay on the rails — however that would be a chunk of ops work. 



![dance3](https://github.com/user-attachments/assets/8fcba512-0cb6-4b0b-b675-5b177b163e93)

Reply:
Nice one! Maybe worth flagging this in slack too ?